LibreNMS/Snmptrap/Handlers/JnxBgpM2Established.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * JnxBgpM2Established.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2018 KanREN, Inc.
+ * @author     Neil Kahle <nkahle@kanren.net>
+ */
+
+namespace LibreNMS\Snmptrap\Handlers;
+
+use App\Models\Device;
+use LibreNMS\Interfaces\SnmptrapHandler;
+use LibreNMS\Snmptrap\Trap;
+use Log;
+
+class JnxBgpM2Established implements SnmptrapHandler
+{
+    /**
+     * Handle snmptrap.
+     * Data is pre-parsed and delivered as a Trap.
+     *
+     * @param Device $device
+     * @param Trap $trap
+     * @return void
+     */
+    public function handle(Device $device, Trap $trap)
+    {
+        
+        $peerState = $trap->getOidData($trap->findOid('BGP4-V2-MIB-JUNIPER::jnxBgpM2PeerState'));
+        $peerAddr = $this->formatIpv6($trap->getOidData($trap->findOid('BGP4-V2-MIB-JUNIPER::jnxBgpM2PeerRemoteAddr.')));
+        Log::event("BGP Peer $peerAddr is now in the $peerState state", $device->device_id, 'trap', 1);
+    }
+
+    public function formatIpv6($ipv6Raw)
```

Comment:
We have a function that does this. `IP::fromHexString()`

Reply:
I'll take a look at it. Is there a way to find to easily see what functions are available in LibreNMS, or is it something you have to pickup?

Reply:
I just search the codebase usually.